module/input/SensorFilter/src/SensorFilter.hpp

```diff
@@ -44,6 +44,39 @@ namespace module::input {
 
         utility::math::filter::UKF<double, MotionModel> motionFilter{};
 
+        struct FootDownMethod {
+            enum Value { UNKNOWN = 0, Z_HEIGHT = 1, LOAD = 2, FSR = 3 };
+            Value value = Value::UNKNOWN;
+
+            // Constructors
+            FootDownMethod() = default;
+            FootDownMethod(int const& v) : value(static_cast<Value>(v)) {}
+            FootDownMethod(Value const& v) : value(v) {}
+            FootDownMethod(std::string const& str) {
+                // clang-format off
+                        if      (str == "Z_HEIGHT") { value = Value::Z_HEIGHT; }
+                        else if (str == "LOAD") { value = Value::LOAD; }
+                        else if (str == "FSR")  { value = Value::FSR; }
+                        else {
+                            value = Value::UNKNOWN;
+                            throw std::runtime_error("String " + str + " did not match any enum for ServoID");
+                        }
+                // clang-format on
+            }
+
+            // Conversions
+            operator Value() const {
```

Comment:
operators can be `[[nodiscard]]` too
```suggestion
            [[nodiscard]] operator Value() const {
```